python-lib/sharepoint_client.py

```diff
@@ -24,11 +24,13 @@ class SharePointClientError(ValueError):
 
 class SharePointClient():
 
-    def __init__(self, config):
+    def __init__(self, config, max_workers=1, batch_size=100):
         self.sharepoint_root = None
         self.sharepoint_tenant = None
         self.sharepoint_url = None
         self.sharepoint_origin = None
+        self.max_workers = max_workers
```

Comment:
In the constructor, two differents notions are intermingled:
- client fields initialization
- remote list deletion/creation, read_schema and columns creation
The last point should be moved to a specific method (prepare_remote_list?) to isolate between client initilisation and remote init.
The call to prepare_remote_list should be done SharePointListsConnector.get_writer